src/components/mermaidMap/SingleSiteMap/SingleSiteMap.js

```diff
@@ -169,19 +174,34 @@ const SingleSiteMap = ({
 
   const updateBenthicLayers = (dataLayerFromLocalStorage) =>
     setGeomorphicOrBenthicLayerProperty(map.current, 'atlas-benthic', dataLayerFromLocalStorage)
-  const placeMarkerButtonText =
-    formLatitudeValue && formLongitudeValue
-      ? language.pages.siteForm.replaceMarker
-      : language.pages.siteForm.placeMarker
 
   const handlePlaceMarkerClick = () => {
     setIsMarkerBeingPlaced(!isMarkerBeingPlaced)
+    hasLatLngChanged && setHasLatLngChanged(false)
   }
 
+  useEffect(
+    function updatePlaceMarkerButtonText() {
+      if (!isMarkerBeingPlaced && (!formLatitudeValue || !formLongitudeValue)) {
+        setPlaceMarkerButtonText(language.pages.siteForm.placeMarker)
+      }
+      if (!isMarkerBeingPlaced && formLatitudeValue && formLongitudeValue) {
+        setPlaceMarkerButtonText(language.pages.siteForm.replaceMarker)
+      }
+      if (isMarkerBeingPlaced && !hasLatLngChanged) {
+        setPlaceMarkerButtonText(language.pages.siteForm.cancelPlaceMarker)
+      }
+      if (isMarkerBeingPlaced && hasLatLngChanged) {
+        setPlaceMarkerButtonText(language.pages.siteForm.done)
```

Comment:
UX question: what if we have changed the location of the marker, but want to cancel? We can only select 'done'.

Reply:
Is cancel and done doing the same operation here, but we are just changing language for the user? Is that creating low-value complexity in the code?

Reply:
good point - maybe we add two buttons , cancel and done

Reply:
I think it's more for the user - if they activate the pin and want to get out, they can click cancel. If they activate it + move it, it would likely be intentional (when done appears)